opcodes/opcodes_dxil_builtins.cpp

```diff
@@ -357,15 +357,33 @@ bool analyze_dxil_instruction(Converter::Impl &impl, const llvm::CallInst *instr
 	{
 		// Mark alloca'd variables which should be considered as payloads rather than StorageClassFunction.
 		auto *payload = instruction->getOperand(15);
-		impl.handle_to_storage_class[payload] = spv::StorageClassRayPayloadKHR;
+		auto storage = impl.get_effective_storage_class(payload, spv::StorageClassFunction);
+		if (storage != spv::StorageClassFunction && storage != spv::StorageClassRayPayloadKHR)
+		{
+			impl.handle_to_storage_class[payload] = spv::StorageClassFunction;
```

Comment:
Should only override storage classes for AllocaInst instructions.

Reply:
Resolved locally.